src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/VectorMath.cs

```diff
@@ -578,19 +814,19 @@ internal static class VectorMath
 
             if (typeof(TVectorInt64) == typeof(Vector64<long>))
             {
-                return (TVectorDouble)(object)Vector64.ConvertToDouble((Vector64<long>)(object)vector);
+                result = (TVectorDouble)(object)Vector64.ConvertToDouble((Vector64<long>)(object)vector);
```

Comment:
What drove the changes here around how the result is returned?

Reply:
Consistency around the intent/way the code was written.

Most compilers tend to perform better and need to do less work with a single return path, the JIT has historically been no exception, and since we need to handle returning in the `throw` path anyways, I had intended for it to be setup like this originally.